designs/SPEC_TEMPLATE.md

```diff
@@ -0,0 +1,48 @@
+
+# Spec Name
+
+* Status: In Review / Reviewed / Implementing / Implemented
+* Author(s): [NuGet](https://github.com/NuGet)
+* Issue: [1](https://github.com/NuGet/Home/issues/1) Issue Title
+
+## Problem Background
+
+Present the background and the motivation for this design. 
+Present evidence why this is an important problem to tackle
+
+## Who are the customers
+
+Customers affected by this change.
+
+## Requirements
+
+Clarify the requirements in a few bullet points
+
+## Goals
+
+Specifically call out the goals for this proposed design.
+
+## Non-Goals
+
+Call out things that are not in consideration for this design. 
+
+## Solution
+
+Describe the proposed solution in as much detail as you deem necessary.

```

Comment:
Do you specify the expected Release here? Or are specs completely independent from implementation schedule?
I know PMs often want to know when to announce things, so not sure if estimation comes into play for a spec or not.

Reply:
Up to the spec author. 
Some specs will discuss this, others won't. 
It really depends on the specific feature, the spec maturity etc.